packages/camera/camera_android_camerax/test/system_services_test.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:camera_android_camerax/src/camerax_library.pigeon.dart'
+    show CameraPermissionsErrorData;
+import 'package:camera_android_camerax/src/system_services.dart';
+import 'package:camera_platform_interface/camera_platform_interface.dart'
+    show CameraException, DeviceOrientationChangedEvent;
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'system_services_test.mocks.dart';
+import 'test_camerax_library.pigeon.dart';
+
+@GenerateMocks(<Type>[TestSystemServicesHostApi])
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('SystemServices', () {
+    tearDown(() => TestProcessCameraProviderHostApi.setup(null));
+
+    test(
+        'requestCameraPermissionsFromInstance completes normally without errors test',
+        () async {
+      final MockTestSystemServicesHostApi mockApi =
+          MockTestSystemServicesHostApi();
+      TestSystemServicesHostApi.setup(mockApi);
+      const bool enableAudio = true;
```

Comment:
nit: I think we typically test against literals instead of variables in this repo

[go/unit-testing-practices](go/unit-testing-practices)